libminifi/src/utils/ThreadPool.cpp

```diff
@@ -182,15 +185,22 @@ void ThreadPool<T>::manageWorkers() {
 
 template<typename T>
 void ThreadPool<T>::start() {
-  if (nullptr != controller_service_provider_) {
-    auto thread_man = controller_service_provider_->getControllerService("ThreadPoolManager");
-    thread_manager_ = thread_man != nullptr ? std::dynamic_pointer_cast<controllers::ThreadManagementService>(thread_man) : nullptr;
-  } else {
-    thread_manager_ = nullptr;
-  }
-
   std::lock_guard<std::recursive_mutex> lock(manager_mutex_);
   if (!running_) {
+    thread_manager_.reset();
+    if (nullptr != controller_service_provider_) {
+      auto service = controller_service_provider_->getControllerService("ThreadPoolManager");
+      if (!service) {
+        logger_->log_info("Could not find a ThreadPoolManager service");
+      } else {
+        if (auto thread_manager_service = std::dynamic_pointer_cast<controllers::ThreadManagementService>(service)) {
+          thread_manager_ = thread_manager_service;
+        } else {
+          logger_->log_error("Found ThreadPoolManager, but it is not a ThreadManagementService");
+        }
+      }
+    }
```

Comment:
I think this would be nicer as
```suggestion
    thread_manager_ = createThreadManager();
```
with the rest of the logic moved to the helper function, which could use a return-early pattern

Reply:
good idea, changed it